internal/providersdkv2/resource_vault_cluster.go

```diff
@@ -1290,6 +1441,54 @@ func getValidObservabilityConfig(config map[string]interface{}) (*vaultmodels.Ha
 		}
 	}
 
+	if httpURI != "" || httpMethod != "" || httpCodec != "" {
+		if observabilityConfig != nil {
+			return nil, tooManyProvidersErr
+		}
+
+		if httpURI == "" || httpMethod == "" || httpCodec == "" {
+			missingParamErr = diag.Errorf("http configuration is invalid: configuration information missing")
+		}
+
+		var httpBearerAuth *vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth
+		var httpBasicAuth *vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth
+
+		if httpBearerToken != "" {
+			httpBearerAuth = &vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth{
+				Token: httpBearerToken,
+			}
+
+			// only one of basic or bearer authentication should be submitted
+			if httpBasicUser != "" || httpBasicPassword != "" {
+				missingParamErr = diag.Errorf("http configuration is invalid: either the basic or bearer authentication method can be submitted, but not both")
+			}
+		} else {
+			// http basic requires both the username and password to be filled
+			if httpBasicUser != "" && httpBasicPassword == "" || httpBasicUser == "" && httpBasicPassword != "" {
+				missingParamErr = diag.Errorf("http configuration is invalid: basic authentication requires username and password")
+			} else {
+				httpBasicAuth = &vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth{
+					User:     httpBasicUser,
+					Password: httpBasicPassword,
+				}
+			}
+		}
```

Comment:
nit: I think this could be refactored to make it easier to understand and avoid this case where we're setting `httpBasicAuth` even if nether `httpBasicUser` nor `httpBasicPassword` are set:
* create helper validation function, smt like `validateHttpAuth`, that allows avoiding nested if's by returning an error early
* in the function first check for token and basic at the same time, return an error
* then check for token and just set `httpBearerAuth`, that will be returned later
* then check for either of the basic auth vars, if either are set check that both are set an return an error if not
* return httpBasicAuth and httpBearerToken, which can both be nil if no auth was configured

that's just an idea tho, but I find the nested ifs a bit hard to follow